### PR TITLE
LUCENE-9907: Remove packedInts dependency on StoredFieldsFormat

### PR DESCRIPTION
**Pull request by iverase**
This PR removes the dependency on PackedInts to the StoredFieldsFormat in favour of directReader / DirectWriter.

### CODE REVIEW

**Review by jpountz — COMMENTED** [review]
I'm not sure if DirectReader is the right tool here since we need bulk decoding here. Maybe we should have a variant of `ForUtil` that only optimizes for numbers of bits per value 0, 8 and 32 or something like that?

**Review by jpountz — COMMENTED** [review]
I think we could further simplify this by not optimizing for 16 or 24 bits per value: these numbers of bits per value would only get used when documents are quite large, so you would unlikely care about saving one more byte per document?

However I wonder if we should look into vectorizing this decoding like postings do. I remember seeing this code path in profilers, which makes sense because even if we can only decompress part of a block thanks to how they are split in 10 and use dictionaries, we still always decode offsets and numbers of stored fields for _all_ documents in the block.

File: lucene/core/src/java/org/apache/lucene/codecs/lucene90/compressing/Lucene90CompressingStoredFieldsReader.java [hunk 1]
```diff
@@ -373,6 +373,7 @@ private SerializedDocument(DataInput in, int length, int numStoredFields) {
 
     private int[] offsets = IntsRef.EMPTY_INTS;
     private int[] numStoredFields = IntsRef.EMPTY_INTS;
+    private StoredFieldsInts intsReader = new StoredFieldsInts();
```
Comment (jpountz) [inline lucene/core/src/java/org/apache/lucene/codecs/lucene90/compressing/Lucene90CompressingStoredFieldsReader.java:376]: let's make it final?

File: lucene/core/src/java/org/apache/lucene/codecs/lucene90/compressing/StoredFieldsInts.java [hunk 2]
```diff
@@ -73,17 +136,59 @@ static void readInts(IndexInput in, int count, int[] values, int offset) throws
     }
   }
 
-  private static void readInts8(IndexInput in, int count, int[] values, int offset)
-      throws IOException {
-    for (int i = 0; i < count; i++) {
-      values[offset + i] = Byte.toUnsignedInt(in.readByte());
+  private void readInts8(IndexInput in, int count, int[] values, int offset) throws IOException {
```
Comment (jpountz) [inline lucene/core/src/java/org/apache/lucene/codecs/lucene90/compressing/StoredFieldsInts.java:139]: Unfortunately, for the JVM to auto-vectorize this, I think we'd need `values` to be a `long[]`.

**Review by jpountz — APPROVED** [review]
Two minor comments, otherwise LGTM!

File: lucene/core/src/java/org/apache/lucene/codecs/lucene90/compressing/StoredFieldsInts.java (hunk 2 above)
Comment (iverase) [reply lucene/core/src/java/org/apache/lucene/codecs/lucene90/compressing/StoredFieldsInts.java:139]: yes, I was thinking that.

File: lucene/core/src/java/org/apache/lucene/codecs/lucene90/compressing/Lucene90CompressingStoredFieldsReader.java (hunk 1 above)
Comment (iverase) [reply lucene/core/src/java/org/apache/lucene/codecs/lucene90/compressing/Lucene90CompressingStoredFieldsReader.java]: I went back to make the method static as now we do not need any helper array

File: lucene/core/src/test/org/apache/lucene/codecs/lucene90/compressing/TestStoredFieldsInt.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import org.apache.lucene.util.TestUtil;
+
+public class TestStoredFieldsInt extends LuceneTestCase {
+
+  public void testRandom() throws Exception {
+    int numIters = atLeast(100);
+    try (Directory dir = newDirectory()) {
+      for (int iter = 0; iter < numIters; ++iter) {
+        int[] values = new int[random().nextInt(5000) + 1];
+        final int bpv = TestUtil.nextInt(random(), 1, 32);
```
Comment (jpountz) [inline lucene/core/src/test/org/apache/lucene/codecs/lucene90/compressing/TestStoredFieldsInt.java:34]: With bpv == 32, `TestUtil.nextInt(random(), 0, (1 << bpv) - 1)` always returns 0.

```suggestion
        final int bpv = TestUtil.nextInt(random(), 1, 31);
```

File: lucene/core/src/test/org/apache/lucene/codecs/lucene90/compressing/TestStoredFieldsInt.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        test(dir, values);
+      }
+    }
+  }
+
+  public void testAllEquals() throws Exception {
+    try (Directory dir = newDirectory()) {
+      int[] docIDs = new int[random().nextInt(5000) + 1];
+      final int bpv = TestUtil.nextInt(random(), 1, 32);
```
Comment (jpountz) [inline lucene/core/src/test/org/apache/lucene/codecs/lucene90/compressing/TestStoredFieldsInt.java:46]:
```suggestion
      final int bpv = TestUtil.nextInt(random(), 1, 31);
```

**Review by jpountz — APPROVED** [review]

**Pull request merged**